File: tutorials/testing/java/tut_array1d_p3.java
import java.util.Iterator;

import bridges.connect.Bridges;
import bridges.base.Array1D;

// Part 3 of this tutorial describes use of iterators with BRIDGES Arrays

public class tut_array1d_p3 {
	public static void main(String[] args) throws Exception {

		//create the Bridges object, set credentials
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");

		// set title
		bridges.setTitle("One-Dimensional Array Example");

		// set description
		bridges.setDescription("Illustrating looping and iterators in 1D Arrays");

		//  set array dimensions, allocate array of elements
		int arraySize = 10;

		Array1D<Integer> arr = new Array1D<Integer> (arraySize);

		// populate the array, with squares of indices

		for (int k = 0; k < arr.getSize(); k++) {
			arr.getElement(k).setValue(k * k);

			// use the array values to label the elements
			arr.getElement(k).setLabel(String.valueOf(arr.getElement(k).getValue()));
		}

		System.out.println ("Printing values of array..");

		// using regular for loops
		System.out.println ("Using regular for loop..");
		for (int k = 0; k < 10; k++)
			System.out.println("\t" + arr.getElement(k).getValue());


		//use an iterator to print the values of the array

		System.out.println("Using an iterator..\n");
		Iterator<Integer> iter = arr.iterator();
		while (iter.hasNext()) {
			System.out.println("\t" + iter.next());
		}

		// using a for-each loop
		System.out.println("Using range loop:");
		for (int  s : arr)
			System.out.println("\t " + s);

		// tell Bridges the  datastructure to visualize
		bridges.setDataStructure(arr);

		// visualize the list
		bridges.visualize();
	}
}

File: tutorials/testing/python/tut_array1d_p3.py
from bridges.bridges import *
from bridges.array import *
from bridges.array1d import *
import sys


def main():
    # Part 3 of this tutorial describes how to use iterators with arrays
    # We will use the same example array 


    # create the Bridges object, set credentials
    bridges = Bridges(ASSIGNMENT_NUMBER, "USER_ID", "API_KEY")


	# set  title 
    bridges.set_title("One-Dimensional Array Example")

    # set description
    bridges.set_description("Illustrates using iterators to print array elements")

    # set array dimensions, allocate array of elements
    arraySize = 10
    arr = Array1D(arraySize)

    # populate the array with squares of indicies
    for i in range(arr.size):
        arr[i].label = i*i

    # print elements of list

    # using a regular while loop
    print("Printing array elements using a regular while loop")
    k = 0
    while (k < 10):
        print(arr[k].label)
        k = k+1
    
    # using an iterator
    array_iterator = iter(arr)
    el = next(array_iterator)
    print("Printing array elements using an iterator")
    while True:
        try:
            el = next(array_iterator)
            print(el.label)
        except StopIteration:
            break 

    # use range loop
    print("Printing array elements using a range loop")
    for el in arr:
        print(el.label)

    # tell Bridges what data structure to visualize
    bridges.set_data_structure(arr)

    # visualize the list
    bridges.visualize()

if __name__ == "__main__":
    main()
